### PR TITLE
Add a new section, When to add or edit functional tests

### PR DESCRIPTION
**Pull request by stevepiercy**
See https://github.com/Pylons/deform/issues/472#issuecomment-686390541

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -88,7 +88,21 @@ We use the following for running functional tests.
...
-If you add or change a feature that reduces test coverage or causes a functional test to fail, then you also must submit a pull request to the [deformdemo](https://github.com/pylons/deformdemo) repository to go along with your functional test change to Deform.
+### When to add or edit functional tests
+
+If you add a feature to Deform, or change a feature in Deform that causes any functional test to fail, then you also must submit a pull request that includes sufficient functional tests to the [deformdemo](https://github.com/pylons/deformdemo) repository paired to your pull request in Deform. Before submitting any pull request, all tests must pass using `tox` in your local development environment.
+
+All pull requests automatically run tests. However when there are paired pull requests in both Deform and deformdemo, we have a chicken-and-egg situation. Each repository cannot pull the version of the other referenced in the pull request to run functional tests. Automated testing will fail for both pull requests.
+
+We work around this as follows.
+
+1. Ensure that all of Deform's automated builds pass, except for the functional test build, for that pull request.
```
[the remark this answers is not in the document]
Comment (stevepiercy) [reply contributing.md:99]: I'm going to need some commands for setting upstream and origin. I use PyCharm for this. I'll be pushing an update shortly.

[the remark this answers is not in the document]
Comment (stevepiercy) [reply contributing.md:122]: OK, I think I captured everything, except for those git commands, which I would like your help on. My git command line is weak.

[the remark this answers is not in the document]
Comment (sydoluciani) [reply contributing.md:122]: cd deform
git remote add upstream https://github.com/Pylons/deform.git
git remote set-url origin git@github.com:git_user_id/deform.git

cd deformdemo_functional_tests
git remote add upstream https://github.com/Pylons/deformdemo.git
git remote set-url origin git@github.com:git_user_id/deformdemo.git

[the remark this answers is not in the document]
Comment (stevepiercy) [reply contributing.md:128]: Got it. Updated. Thank you!

[the remark this answers is not in the document]
Comment (sydoluciani) [reply contributing.md:128]: Did you have a chance to test it locally based on document and setting those environment variables to see if local firefox and selenium firefox container works ? 

Connecting to container using VNC left to view the progress of tests in Selenium container but I will submit a new pull request when it is ready and I will fix any indentation on contributing.md file at the sime time.

You can merge both these pull requests.

[the remark this answers is not in the document]
Comment (sydoluciani) [reply contributing.md:128]: These pull requests only affect local testing, and shall not affect tests running in Travis, so Travis test is not enough.
I already test it but it is good to have another local test done to make sure.

Try the latest pull requests on both Deform and Deformdemo, or you can merge and the pull and test it locally.

[the remark this answers is not in the document]
Comment (stevepiercy) [reply contributing.md:128]: @sydoluciani I tried them out, and it went pretty smoothly. Thank you!

I will delay merging this PR until your pair of PRs, https://github.com/Pylons/deformdemo/pull/84 and https://github.com/Pylons/deform/pull/473, are merged. I will rebase this PR when those two are merged.

**Pull request merged**